### ISSUE TITLE
`SymmetryOperation_PET_CartesianGrid` are not correctly reverting `basic_bin` back to the original `bin`

### ISSUE THREAD

**Issue by robbietuk** [UCL/STIR#999]
To summarise, 

1. The symmetry methodology takes a `bin`, finds the basic bin with [`find_basic_bin()`](https://github.com/UCL/STIR/blob/06963f9292d7e9cbfe5b3ffa8d49eb7a533ec6eb/src/include/stir/recon_buildblock/DataSymmetriesForBins_PET_CartesianGrid.inl#L474-L666) and the corresponding [symmetry operation](https://github.com/UCL/STIR/blob/06963f9292d7e9cbfe5b3ffa8d49eb7a533ec6eb/src/include/stir/recon_buildblock/DataSymmetriesForBins_PET_CartesianGrid.inl#L144-L435) to revert `basic_bin` back to `bin`.
 
2. Generally `basic_bin` is used to check against the cache or the `probabilities`/lor is computed, see [calculate_proj_matrix_elems_for_one_bin()](https://github.com/UCL/STIR/blob/06963f9292d7e9cbfe5b3ffa8d49eb7a533ec6eb/src/include/stir/recon_buildblock/ProjMatrixByBin.inl#L41-L104)

3. The symmetry operation (found in 1.) is used to compute the `probabilities` elements (values on the cartesian grid) that correspond to `bin`s LOR (_this bit works_) and convert the`basic_bin` (stored in the `probabilities` object) back to `bin` (_this bit is broken_). 

___

I added an additional test into the [`test_DataSymmetriesForBins_PET_CartesianGrid`](https://github.com/UCL/STIR/blob/06963f9292d7e9cbfe5b3ffa8d49eb7a533ec6eb/src/recon_test/test_DataSymmetriesForBins_PET_CartesianGrid.cxx#L125-L127) code by replacing the `if` statement 
```
if (!check(elems_no_sym == elems_with_sym, "comparing lors")) 
```
with
```
      if (!check(elems_no_sym == elems_with_sym, "comparing lors") ||
          !check(elems_with_sym.get_bin() == elems_no_sym.get_bin(), "Comparing Bin"))
```
An example set of errors (with some further `std::cerr` modifications) are
```
Error. Comparing Bin
Current bin: 	segment = -2, 	axial pos 13, 	view = 0, 	tangential_pos_num = 3
Symm bin: 	segment = 2, 	axial pos 0, 	view = 0, 	tangential_pos_num = 3

Error. Comparing Bin
Current bin: 	segment = -2, 	axial pos 13, 	view = 0, 	tangential_pos_num = 6
Symm bin: 	segment = 2, 	axial pos 0, 	view = 0, 	tangential_pos_num = 6

Error. Comparing Bin
Current bin: 	segment = -2, 	axial pos 0, 	view = 1, 	tangential_pos_num = 0
Symm bin: 	segment = 2, 	axial pos 0, 	view = 1, 	tangential_pos_num = 0

```

As far as I can tell, there are only issues with the flipping of `segments` and `axial_positions` offsets. 

___
This issue was observed in https://github.com/NikEfth/STIR/pull/25 but this confirms that it exists in the non-TOF STIR/master.

**Comment by KrisThielemans** [comment]
interesting... Still the case after #1000?

**Comment by robbietuk** [comment]
Unfortunately so. It seems there are problems with the transformations of bin.

**Comment by robbietuk** [comment]
Additionally, we are no testing any rotational symmetries due to the intrinisic tilt of the E953 scanner. PR incoming.

**Comment by KrisThielemans** [comment]
> Additionally, we are no testing any rotational symmetries due to the intrinisic tilt of the E953 scanner. PR incoming.

indeed. these were tested before we enabled view-offset, but now not anymore.

**Comment by robbietuk** [comment]
Perhaps this is kept open? The rotational symmetries have not been verified yet.

**Comment by KrisThielemans** [comment]
I think this will be fine, so prefer to keep it closed, until we find that I'm wrong..
